x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToDatePeriodSerializationTests.java

```diff
@@ -7,13 +7,32 @@
 
 package org.elasticsearch.xpack.esql.expression.function.scalar.convert;
 
-import org.elasticsearch.xpack.esql.core.expression.Expression;
-import org.elasticsearch.xpack.esql.core.tree.Source;
-import org.elasticsearch.xpack.esql.expression.AbstractUnaryScalarSerializationTests;
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.xpack.esql.expression.AbstractExpressionSerializationTests;
+
+import java.io.IOException;
+
+import static org.elasticsearch.xpack.esql.EsqlTestUtils.randomLiteral;
+import static org.elasticsearch.xpack.esql.core.type.DataType.KEYWORD;
+
+public class ToDatePeriodSerializationTests extends AbstractExpressionSerializationTests<ToDatePeriod> {
+    @Override
+    protected ToDatePeriod createTestInstance() {
+        return new ToDatePeriod(randomSource(), randomLiteral(KEYWORD));
+    }
+
+    @Override
+    protected ToDatePeriod mutateInstance(ToDatePeriod instance) throws IOException {
+        return null;
```

Comment:
Not sure how much value we gain from having the serialization tests since we don't serialize this. Not mutating seems off to me, though; there are still tests here that can mutate and check `equals` and `hashCode`. `mutate` could replace the contained literal with another, random literal (using `randomValueOtherThan`).

Reply:
+1
If there is a reason to have this (maybe I am missing it), it would be good to know.

Reply:
This is a valid question and I asked myself too when I added these tests, and also consulted @nik9000. I think Nik has a point, at least convinced me, that this situation might be similar to e(), pi(), the things in common is that they don't need to be serialized, however the tests drive doc generation. Let me discuss this with Nik again to see the best options here.

Reply:
Thanks Nik! I'll remove these two serialization tests, today, data node won't see these two functions, they have to be folded by coordinator.